crates/sparrow-runtime/src/prepare/error.rs

```diff
@@ -1,15 +1,7 @@
-use std::path::PathBuf;
-
 use url::Url;
 
-use crate::stores::ObjectStoreUrl;
-
 #[derive(derive_more::Display, Debug)]
 pub enum Error {
-    #[display(fmt = "unable to open '{path:?}'")]
-    OpenFile { path: PathBuf },
```

Comment:
Since the error enums are public, we don't get a warning / lint to remove errors that are no longer used. That kind of sucks. On the other hand, since they are potentially part of the public API, removing them is actually a potential problem.

I suspect we may want a pattern where we have top-level, public error enums with more general cases ("InvalidArguments", "Internal", etc.) and then we can have internal (`pub crate`) error enums with more details.

That way the `pub enum` is the "public" contract, and we've set ourselves up for not needing to change that as often.